### DISCUSSION TITLE
How to get the payload data?

### DISCUSSION THREAD

**Discussion by seprim** [eModbus/eModbus#88] [eModbus applications]
Maybe it's a stupid question but I'm extremely stumped. Unfortunately, I can't figure out how to read the payload with the library.
So far the examples work, but I can't get any further.
Can anyone give me a hint?

**Comment by Miq1** [comment]
In any case you will want to start with the ModbusMessage you have received - either request (as a server) or response (as a client). If you have a look at the documentation you will see the ``get()`` family of calls that will extract any data format from the ModbusMessage at a given spot.

In the server examples, the ``FC03`` callback functions f.i. are using the ``get()`` calls to extract address and number of words.
